### PR TITLE
100 add documentation for the solver

### PR DESCRIPTION
**Pull request by samvanderzwan**
Added first level of documentation. It is not yet at the highest level, but it is a starting point to build on.

### CODE REVIEW

File: doc/solver/solver_main.rst [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+=========================
+To solve one time step of the simulation the solver is used.
+This solver is solving both the hydraulic and thermo dynamic equations.
+As unknown the following quantities are solved:
+
+#. Mass flow rate [kg/s]
+#. Total pressure [Pa]
+#. Specific internal energy [J/kg]
+
+The later is depending on the temperature and specifies the amount of energy needed to change the
```
Comment (MichielTukker) [inline doc/solver/solver_main.rst:11]: Typo :  the **latter** is depending on....

File: doc/solver/solver_utility.rst [hunk 2]
```diff
@@ -0,0 +1,8 @@
+.. _solver-utility:
+
+Solver utility functions
++++++++++++++++++++++++++++++++++++++
+To be filled later
```
Comment (MichielTukker) [inline doc/solver/solver_utility.rst:5]: List functions and use their docstring for documentation.  If there are no "public" utility functions,  remove the section

**Review by MichielTukker — CHANGES_REQUESTED** [review]
Typos! 

Also:  why create paragraphs of text in external documentation files.  Maybe we should add the text from the RST files in the respective docstrings for each class,  and just display the autodoc output as documentation.  Keeps everything in a single location

File: ci/win32/__init__.py
[the remark this answers is not in the document]
Comment (samvanderzwan) [reply ci/win32/__init__.py:1]: done

File: cluster3_spars_year
[the remark this answers is not in the document]
Comment (samvanderzwan) [reply cluster3_spars_year:1]: hmmm file that should not have been committed same as the other, is deleted now

File: doc/controller/controller.rst [hunk 3]
```diff
@@ -0,0 +1,3 @@
+Controller class
+=====================================
+To be filled.
```
[the remark this answers is not in the document]
Comment (samvanderzwan) [reply doc/controller/controller.rst:3]: Documentation is now only for the solver when we extend the controller class #134 then we will also add documentation there.

File: doc/solver/assets/base_asset.rst [hunk 4]
```diff
@@ -0,0 +1,14 @@
+.. _base-asset:
+
+Base asset class
+++++++++++++++++++++++++
+Base asset class, which is used to add the first basic equations for the general assets.
+This are equation to set the temperature or pressure or an equation to set the pressure or
+temperature
```
[the remark this answers is not in the document]
Comment (samvanderzwan) [reply doc/solver/assets/base_asset.rst:7]: fixed

File: doc/solver/assets/base_asset.rst [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+equal to the pressure of the connected node. This class is split from the bast item class to avoid
```
[the remark this answers is not in the document]
Comment (samvanderzwan) [reply doc/solver/assets/base_asset.rst:8]: fixed

File: doc/solver/assets/solver_pipe.rst [hunk 6]
```diff
@@ -0,0 +1,10 @@
+.. _solver-pipe:
+
+Solver pipe class
+++++++++++++++++++++++++
+The pipe class is a solver asset that represents a pipe in the network.
+It is used to calculate the flow in the pipe and the pressure drop across the pipe.
+The heat loss is also calculated based on the insulation of the pipeline.
+
```
[the remark this answers is not in the document]
Comment (samvanderzwan) [reply doc/solver/assets/solver_pipe.rst:8]: Added some explanation

File: doc/solver/assets/solver_producer.rst [hunk 7]
```diff
@@ -0,0 +1,12 @@
+.. _solver-producer:
+
+Solver producer class
+++++++++++++++++++++++++
+The Solver producer class can be used to represent a producer or a consumer in the network.
+This has been chosen since a consumer is the same as an producer onl the heat flow is in the other
```
[the remark this answers is not in the document]
Comment (samvanderzwan) [reply doc/solver/assets/solver_producer.rst:6]: fixed

File: doc/solver/assets/solver_producer.rst [hunk 8]
```diff
@@ -0,0 +1,12 @@
+.. _solver-producer:
+
+Solver producer class
+++++++++++++++++++++++++
+The Solver producer class can be used to represent a producer or a consumer in the network.
```
[the remark this answers is not in the document]
Comment (samvanderzwan) [reply doc/solver/assets/solver_producer.rst:5]: good point, but then we also need to rename it in code. I made new issue to discuss this: #135

File: doc/solver/solver_utility.rst (hunk 2 above)
Comment (samvanderzwan) [reply doc/solver/solver_utility.rst:5]: Done

**Comment by samvanderzwan** [comment]
In would not only use autoclass, This is how it nows looks and that is somewhat better then if you would only use autoclas:

File: doc/solver/solver_main.rst (hunk 1 above)
Comment (samvanderzwan) [reply doc/solver/solver_main.rst]: fixed

**Comment by samvanderzwan** [comment]
Fixed all issues, @MichielTukker can you check?

File: doc/controller/controller.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (MichielTukker) [reply doc/controller/controller.rst:3]: Created issue for that, See #136

File: doc/solver/assets/solver_pipe.rst [hunk 9]
```diff
@@ -0,0 +1,11 @@
+.. _solver-pipe:
+
+Solver pipe class
+++++++++++++++++++++++++
+The pipe class is a solver asset that represents a pipe in the network.
+It is used to calculate the flow in the pipe and the pressure drop across the pipe.
+It is derived from the fall type class, and extended with the heat loss over the pipeline.
+The heat loss is calcualted based on the actual physical properties of the pipe.
```
Comment (MichielTukker) [inline doc/solver/assets/solver_pipe.rst:8]: Typo:   **calcualted**

File: doc/solver/equation_object.rst [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+
+The index is the sum of the index in the matrix, which is obtained by the assets by setting their unknowns in the matrix with the quantity to be solved.
+This can be easily obtained by making use of the CoreQuantIndex class.
+The coefficients are the coefficients which are obtained by linearizing the equations. The same holds for the right hand side. This linearization should be done via the Newton-Raphson methode.
+More information can be found on this `website`_.
+
+.. autoclass:: simulator_core.solver.matrix.equation_object.EquationObject
+   :members:
+
+.. _website: https://en.wikipedia.org/wiki/Newton%27s_method
```
Comment (MichielTukker) [inline doc/solver/equation_object.rst:19]: I think something went wrong with formatting?  (link is separate from text).   Or is this a reference to a footnote/reference list?

**Review by MichielTukker — CHANGES_REQUESTED** [review]

File: doc/solver/assets/solver_pipe.rst (hunk 9 above)
Comment (samvanderzwan) [reply doc/solver/assets/solver_pipe.rst:8]: fixed

File: doc/solver/equation_object.rst (hunk 10 above)
Comment (samvanderzwan) [reply doc/solver/equation_object.rst:19]: In this way the text in the html becomes website and not the full website path. So it is better readable
image

File: doc/solver/assets/base_asset.rst [hunk 11]
```diff
@@ -0,0 +1,14 @@
+.. _base-asset:
+
+Base asset class
+++++++++++++++++++++++++
+Base asset class, which is used to add the first basic equations for the general assets.
+This are equation to set the temperature or pressure or an equation to set the pressure or
```
[the remark this answers is not in the document]
Comment (samvanderzwan) [reply doc/solver/assets/base_asset.rst]: Rewrote this part

**Comment by samvanderzwan** [comment]
Fixed final comments @MichielTukker  can you do final check?

**Review by MichielTukker — APPROVED** [review]

**Pull request merged**